### ISSUE TITLE
Retry for IAM Eventual Consistency in aws_codebuild_project

### ISSUE THREAD

**Issue by bflad** [terraform-providers/terraform-provider-aws#4233]
### Terraform Version

```
Terraform v0.11.7
+ provider.aws v1.14.1
```

### Affected Resource(s)

- aws_codebuild_project

### Terraform Configuration Files

```hcl
resource "aws_codebuild_project" "example" {
  service_role  = "${aws_iam_role.example.arn}"
  # ... other configuration ....
}
```

### Expected Behavior

Created CodeBuild project without error.

### Actual Behavior

```
=== RUN   TestAccAWSCodeBuildProject_basic
--- FAIL: TestAccAWSCodeBuildProject_basic (16.99s)
    testing.go:518: Step 0 error: Error applying: 1 error(s) occurred:
        
        * aws_codebuild_project.foo: 1 error(s) occurred:
        
        * aws_codebuild_project.foo: [ERROR] Error updating CodeBuild project (arn:aws:codebuild:us-west-2:*******:project/test-project-zlgi1d4pe0): InvalidInputException: CodeBuild is not authorized to perform: sts:AssumeRole on arn:aws:iam::*******:role/codebuild-role-zlgi1d4pe0
            status code: 400, request id: 4d80c314-41ff-11e8-9a18-93bbec3717da
```

### Steps to Reproduce

Occasionally:
1. `make testacc TEST=./aws TESTARGS='-run=TestAccAWSCodeBuildProject_basic'`

**Comment by errriclee** [comment]
Now #4238

**Comment by bflad** [comment]
The fix for this has been merged into master and will release with v1.15.0 of the AWS provider, likely tomorrow.
